### PR TITLE
desktop: add parameters to specify window size

### PR DESCRIPTION
**Pull request by CounterPillow**
This adds two optional parameters, --width and --height, which allow the user to specify a desired physical window width and height to be used. If only one of the two parameters is present, the other will be deduced from it and the movie's aspect ratio.

If neither are present, the window's size is set to the movie's logical size, in accordance with the behaviour before this change.

### CODE REVIEW

**Comment by CounterPillow** [comment]
force-pushed as I had to fix the inevitable integer division goof-up.

**Comment by Herschel** [comment]
My first thought is that this is better as settings in a config file instead of CLI options. Do you have a specific use case in mind? Is it that you want to a fullscreen or maximized view on startup?

**Comment by CounterPillow** [comment]
My use case it to be able to specify a window size for specific flashes so that I can then capture them using e.g. OBS.

**Comment by Herschel** [comment]
Do you always set it to an exact size (say, 1920x1080), or do you change the size depending on the SWF? (Trying to decide on the best way to save these settings).

**Comment by CounterPillow** [comment]
Depending on the situation, which also means the SWF in this case. Say I have a fun SWF loop I want to have as a "start" screen of the stream, which I probably want to run at the output size because I need the screen real estate for other things. However, if I were to play any other SWF file, I'd rather do so without editing my config file inbetween runs of ruffle first.

Ruffle also has a command line option to change the rendering quality and the graphics backend, are those not even more appropriate for config files than window size?

I mostly come from an mpv background where anything that goes in a config file can also be set on the command line, and in that case, overrides what was set in the config file.

**Comment by Herschel** [comment]
Yes, in fact all of the options should be available a config file; the issue is simply that we have not implemented a config file yet. 😄 

My vision is that there will be "global" settings, and then also the ability for "per-SWF" settings that override the globals. So in this case, assuming you are capturing Ruffle output for a stream, you could set the settings for a particular SWF.

I don't think every config option will be exposed on the command-line, because there will be a large amount of them, and will also involve structured data (input bindings, for example). Not necessarily opposed to adding width/height, as these are relatively basic options; but trying to think about the long-term plan.

File: desktop/src/main.rs [hunk 1]
```diff
@@ -216,7 +223,21 @@ fn run_player(opt: Opt) -> Result<(), Box<dyn std::error::Error>> {
...
+            * (f64::from(opt.height.unwrap_or(movie_size.height)) / f64::from(movie_size.height)))
+        .round() as u32,
+    );
+    let window_height = opt.height.unwrap_or(
+        (f64::from(movie_size.height)
+            * (f64::from(opt.width.unwrap_or(movie_size.width)) / f64::from(movie_size.width)))
+        .round() as u32,
+    );
+    let window_size = PhysicalSize::new(window_width, window_height);
+    window.set_inner_size(window_size);
```
Comment (Herschel) [inline desktop/src/main.rs:240]: Let's make `movie_size` use `f64`:
`LogicalSize::new(movie.width(), movie.height()).cast::<f64>().to_physical(window.scale_factor())`
Then all of the `f64::from` goes away. winit takes care of the rounding when we call `set_inner_size`

**Comment by CounterPillow** [comment]
> My vision is that there will be "global" settings, and then also the ability for "per-SWF" settings that override the globals. So in this case, assuming you are capturing Ruffle output for a stream, you could set the settings for a particular SWF, and that would be remembered across runs.

This sounds fine; though there is the question of how per-file options are matched to files. If you match them with a full path, moving a file will make it no longer apply those settings (though that's not too bad). If you match just the file name, you get collisions for common files (`player.swf`). Matching by file hash or some such seems unpleasant. Maybe a pattern match so the user can decide themselves how much of the path the match needs for context would be best.

> I don't think every config option will be exposed on the command-line, because there will be a large amount of them

ffmpeg and mpv do both have a huge amount of command line options and sort of cope with it, with the drawback that `-h` is next to no information aside from how to get more help output for all the options. The benefit is that you don't really need to differentiate between command line and config options; when you add a new option, you just add it and it appears in both contexts with perfect consistency with regards to naming. For people who want to enable a specific set of options with just one switch, mpv has "profiles" which are sections of the config file which look like this:
```
[profilename]
foo=bar
baz=1.0
```
and can then be applied with `--profile=profilename`

I've made the float casting changes, and it still seems to work. Though rustfmt did some weird things.

**Comment by Herschel** [comment]
Thank you!

**Comment by CounterPillow** [comment]
Actually, I come bearing bad news: the `cast::<f64>` change broke `ruffle_desktop --width 100`, which now results in a backtrace:
```
[2021-05-06T23:36:00Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadValue (integer parameter out of range for operation)",
        error_code: 2,
        request_code: 12,
        minor_code: 0,
    }
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: XError { description: "BadValue (integer parameter out of range for operation)", error_code: 2, request_code: 12, minor_code: 0 }', /home/fratti/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/window.rs:1006:14
stack backtrace:
   0: rust_begin_unwind
             at /build/rust/src/rustc-1.51.0-src/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /build/rust/src/rustc-1.51.0-src/library/core/src/panicking.rs:92:14
   2: core::option::expect_none_failed
             at /build/rust/src/rustc-1.51.0-src/library/core/src/option.rs:1300:5
   3: core::result::Result<T,E>::unwrap
             at /build/rust/src/rustc-1.51.0-src/library/core/src/result.rs:1037:23
   4: winit::platform_impl::platform::x11::window::UnownedWindow::inner_size_physical
             at /home/fratti/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/window.rs:1003:9
   5: winit::platform_impl::platform::x11::window::UnownedWindow::inner_size
             at /home/fratti/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/linux/x11/window.rs:1011:9
```

This worked with my `from` mess, but I don't know enough rust to know how to fix this appropriately.

**Pull request merged**